packages/xchain-binance/src/client.ts

```diff
@@ -223,59 +226,80 @@ class Client implements BinanceClient, XChainClient {
     }
   }
 
-  getTransactions = async (params?: TxHistoryParams): Promise<TxsPage> => {
-    const clientUrl = `${this.getClientUrl()}/api/v1/transactions`
-    const url = new URL(clientUrl)
+  private searchTransactions = async (params?: { [x: string]: string | undefined }): Promise<TxsPage> => {
+    try {
+      const clientUrl = `${this.getClientUrl()}/api/v1/transactions`
+      const url = new URL(clientUrl)
 
-    url.searchParams.set('address', params ? params.address : this.getAddress())
-    if (params && params.limit) {
-      url.searchParams.set('limit', params.limit.toString())
-    }
-    if (params && params.offset) {
-      url.searchParams.set('offset', params.offset.toString())
-    }
-    if (params && params.startTime) {
-      url.searchParams.set('startTime', params.startTime.toString())
-    }
+      for (const key in params) {
+        const value = params[key]
+        if (value) {
+          url.searchParams.set(key, value)
+        }
+      }
 
-    try {
       const txHistory = await axios.get<BinanceTxPage>(url.toString()).then((response) => response.data)
 
       return {
         total: txHistory.total,
         txs: txHistory.tx.reduce((acc, tx) => {
-          const asset = assetFromString(`${AssetBNB.chain}.${tx.txAsset}`)
-
-          if (!asset) return acc
-
-          return [
-            ...acc,
-            {
-              asset,
-              from: [
-                {
-                  from: tx.fromAddr,
-                  amount: assetToBase(assetAmount(tx.value, 8)),
-                },
-              ],
-              to: [
-                {
-                  to: tx.toAddr,
-                  amount: assetToBase(assetAmount(tx.value, 8)),
-                },
-              ],
-              date: new Date(tx.timeStamp),
-              type: getTxType(tx.txType),
-              hash: tx.txHash,
-            },
-          ]
+          const txData = parseTx(tx)
+
+          return [...acc, ...(txData ? [txData] : [])]
```

Comment:
just IMHO it would be simplier to read smth like this
```ts
return {
        total: txHistory.total,
        txs: txHistory.tx.map(parseTx).filter(Boolean) as as Txs[]
      }
```